schema-derivatives-perpfutures.graphql

```diff
@@ -0,0 +1,1425 @@
+  logIndex: Int!
+
+  " The protocol this transaction belongs to "
+  protocol: DerivPerpProtocol!
+
+  " Address that received the tokens "
+  to: String!
+
+  " Address that sent the tokens "
+  from: String!
+
+  " Block number of this event "
+  blockNumber: BigInt!
+
+  " Timestamp of this event "
+  timestamp: BigInt!
+
+  " Token deposited into pool "
+  tokenIn: Token!
+
+  " Amount of token deposited into pool in native units "
+  amountIn: BigInt!
+
+  " Amount of token deposited into pool in USD "
+  amountInUSD: BigDecimal!
+
+  " Token withdrawn from pool "
+  tokenOut: Token!
+
+  " Amount of token withdrawn from pool in native units "
+  amountOut: BigInt!
+
+  " Amount of token withdrawn from pool in USD "
+  amountOutUSD: BigDecimal!
+
+  " The pool involving this transaction "
+  pool: LiquidityPool!
+}
+
+# An account is a unique Ethereum address
+# Helps to accumulate total unique users
+type Account @entity{
+  " Address of the account "
+  id: Bytes!
+
+    " All premium paid to enter a position. "
+  cumulativeEntryPremiumUSD: BigDecimal!
+
+  " All premium paid to exit a position. "
+  cumulativeExitPremiumUSD: BigDecimal!
+
+  " All premium paid. "
+  cumulativeTotalPremiumUSD: BigDecimal!
+
+  " All premium paid to deposit in liquidity pools. "
+  cumulativeDepositPremiumUSD: BigDecimal!
+
+  " All premium paid to withdraw from liquidity pools. "
+  cumulativeWithdrawPremiumUSD: BigDecimal!
+
+  " All premium paid to enter and exit liquidity pool. "
+  cumulativeTotalLiquidityPremiumUSD: BigDecimal!
+
+   " Total number of long positions "
+  longPositionCount: Int!
+
+  " Total number of short positions "
+  shortPositionCount: Int!
+
+    " Total number of open positions "
+  openPositionCount: Int!
+
+   " Total number of closed positions "
+  closedPositionCount: Int!
+
+  " Total number of liquidations "
+  cumulativeUniqueLiquidatees: Int!
+
+   " All deposit (add liquidity) events occurred in this account "
+  deposits: [Deposit!]! @derivedFrom(field: "account")
+
+  " All withdraw (remove liquidity) events occurred in this account "
+  withdraws: [Withdraw!]! @derivedFrom(field: "account")
+
+  " All Position events occurred in this account "
+  positions: [Position!]! @derivedFrom(field: "account")
+
+}
+
+
+# Helper entity for calculating daily/hourly active users
+type ActiveAccount @entity{
+  " { daily/hourly }-{ Address of the account }-{ Days/hours since Unix epoch } "
+  id: Bytes!
+
+   hourlyActiveUsers: Int!
+  dailyActiveUsers: Int!
+}
+
+# A position is defined as who has what position in the protocol and distribution of longs and shorts at a positional level
+type Position @entity {
+  " { Account address }-{ Market address }-{ Position Side }-{ Optional: Interest Rate Type}-{ Counter } "
+  id: ID!
+
+  " Account that owns this position "
+  account: Account!
+
+  " The market in which this position was opened "
+  market: Market!
```

Comment:
In this case it would be liquidityPool instead of Market?

Reply:
done